kernel/environ.ml

```diff
@@ -405,6 +405,14 @@ let vars_of_global env constr =
     (** FIXME: is Proj missing? *)
     | _ -> raise Not_found
 
+let vars_of_global_reference env gr =
+  let open Globnames in
+  match gr with
+  | VarRef id -> Id.Set.singleton id
+  | ConstRef kn -> lookup_constant_variables kn env
+  | IndRef ind -> lookup_inductive_variables ind env
+  | ConstructRef cstr -> lookup_constructor_variables cstr env
+
```

Comment:
Globnames is in library. OCaml apparently knows how to reorder files, but I would rather either move global_reference to Names or replace the call to this function by a composition of vars_of_global and constr_of_global.